cli/pkg/cmdutil/githelper.go

```diff
@@ -0,0 +1,200 @@
+package cmdutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/rilldata/rill/admin/client"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/sync/semaphore"
+)
+
+var (
+	gitignoreHasDotenvRegexp       = regexp.MustCompile(`(?m)^\.env$`)
+	gitignoreHasDotRillCloudRegexp = regexp.MustCompile(`(?m)^\s*\.rillcloud/`)
```

Comment:
I don't think `.gitignore` requries a trailing slash? I.e. `.rillcloud` is also valid, doesn't have to be `.rillcloud/`